### ISSUE TITLE
How to run a cusparse code?

### ISSUE THREAD

**Issue by LiyanChen-SJTU** [gpgpu-sim/gpgpu-sim_distribution#263]
Hi everybody, 

I'm trying to run a simple cusparse code on GPGPUsim. But when I try to execute it, I'm obtaining Segmentation fault (core dumped). I'm using Ubuntu 18.04 and CUDA 11.1 version.

ptx_ir.cc:303: void symbol_table::set_label_address(const symbol*, unsigned int): Assertion `i != m_symbols.end()' failed. Aborted (core dumped)”.

Then I switch to CUDA 10.0, but get another bug:

GPGPU-Sim: ERROR while parsing output of ptxas (used to capture resource usage information)

Codes can be run on real GPU card.

Thanks in advance!

**Comment by mattsinc** [comment]
Yes, this is a well known issue.  Essentially, after CUDA 8, NVIDIA stopped embedded the PTX implementations of their kernels into CUDA libraries.  Standalone GPGPU-Sim relies on this support, which means newer versions of CUDA like the ones you are trying won't work.  You have two options:

1.  (recommended) Use Accel-Sim (https://github.com/accel-sim/) instead.  Accel-Sim integrates onto the "front end" of GPGPU-Sim (i.e., GPGPU-Sim still runs under the hood), but it has support to extract the implementation of kernels in libraries like cuSparse which allow it to run the latest and greatest CUDA versions with library calls like these.  In my experience, once you get Accel-Sim setup the first time, it's very similar to running standalone GPGPU-Sim.
2.  (not recommended) you can downgrade your version of CUDA to CUDA 8 and run as is, essentially replicating the methodology from this paper: https://arxiv.org/abs/1811.08933

Hope this helps,
Matt

**Comment by LiyanChen-SJTU** [comment]
> 1. Accel-Sim integrates onto the "front end" of GPGPU-Sim (i.e., GPGPU-Sim still runs under the hood), but it has support to extract the implementation of kernels in libraries like cuSparse which allow it to run the latest and greatest CUDA versions with library calls like these.  In my experience, once you get Accel-Sim setup the first time, it's very similar to running standalone GPGPU-Sim.

Thanks a lot. I fix it using Accel-Sim!
